### PR TITLE
Small frontend cleanups

### PR DESCRIPTION
**Pull request by janfaracik**
The intention of this PR is to remove unnecessary and deprecated files to make development easier on Jenkins going forward and to also improve the user experience by cutting down on page file sizes.

* Renames `base-styles-v2.less` to `styles.less` _(renamed just to make it obvious that this is the main CSS file)_
* Removes `ui-refresh-overrides.less` - _(this was an earlier attempt at redesigning parts of Jenkins - unsure if its still wanted - happy to undo removal if so)_
  * Small CSS var cleanup
* Removes `html.jelly` _(deprecated since 2017, only potential uses of it are in private plugins, none in jenkinsci organisation)_
* Removes `svgxuse` polyfill for IE11 _(IE11 is now deprecated so theres no need for the polyfill or custom JS)_
* Removes `msie.js` 

### Proposed changelog entries

* N/A

#### Plugin developers

* `html.jelly` template has been removed - use the more feature rich `layout.jelly` template for your pages 

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — COMMENTED** [review]
The motivation behind this pull request is not clear to me. Therefore I am adding the `needs-motivation` label. Please provide a clear description of the high-level problem this pull request is trying to solve. Recall that while anyone in the community can see this pull request, not everyone necessarily has the same context as you. Ideally, the problem statement would be written in terms of a specific symptom that affects users or the development team. The problem statement should not be written in terms of the solution. Providing a clear description of the problem you are trying to solve and the solution you have chosen helps us evaluate the pull request.

> if this is still needed

Your usage of the subordinating conjunction "if" indicates you are not sure whether or not this is still needed. The correct course of action, then, is to determine the answer to this question before the code is merged (by doing a code search) rather than after the code is merged (by causing a regression). See [the Backwards Compatibility section of JEP-227](https://github.com/jenkinsci/jep/blob/master/jep/227/README.adoc#backwards-compatibility=) for examples of how to search for API usages in sources and in binaries.

File: war/src/main/less/abstracts/theme.less
Comment (NotMyFault) [inline war/src/main/less/abstracts/theme.less:70]: This is is used in the dark theme plugin, it would make sense to remove it there as well then, the other removals of this block are commented out in the dark theme and solarized plugin or unused.

**Review by NotMyFault — COMMENTED** [review]
There are a few isMSIE checks left calling the corresponding - now removed - script. I'd vouch for removing these references as well.

> The motivation behind this pull request is not clear to me.

This is more of a chore polishing components recently touched and can easily go with `skip-changelog`.
IE compatibility has been dropped from the compatibility matrix, hence we no longer need to hack our way around it.

**Comment by janfaracik** [comment]
> There are a few isMSIE checks left calling the corresponding - now removed - script. I'd vouch for removing these references as well.

There's a few in `html.jelly` - wondering if it's worth just removing this file altogether seeing as it was deprecated as of 2017?

**Comment by NotMyFault** [comment]
> > There are a few isMSIE checks left calling the corresponding - now removed - script. I'd vouch for removing these references as well.
> 
> There's a few in `html.jelly` - wondering if it's worth just removing this file altogether seeing as it was deprecated as of 2017?

According to cs.github it has no usages within the jenkinsci org, but we need an upgrade guide for private plugins. 
A one-liner pointing to the feature-rich alternative l:layout, as outlined, should be sufficient.

**Review by basil — CHANGES_REQUESTED** [review]
Thank you very much for providing the motivation for each change! This is extremely helpful context and allows us to do a meaningful review of the PR.

This Git branch is named `small-cleanup`, but it is not a small cleanup. Rather, it is four different cleanups. I suggest splitting such changes into _separate_ PRs, each one covering a distinct and logically self-contained unit of work. As a courtesy, I have taken the liberty of splitting this branch into four logically self-contained units of work:

- [`base-styles-v2`](https://github.com/basil/jenkins/tree/base-styles-v2): basil/jenkins@19229cc7e2803f3945a72529a5107d812a7c2dd3
- [`html-jelly`](https://github.com/basil/jenkins/tree/html-jelly): basil/jenkins@6defa80e8d91b9ce52c43f7a4a95fc29196c91f4
- [`msie-svgxuse`](https://github.com/basil/jenkins/tree/msie-svgxuse): basil/jenkins@bc2bfd4cc33c4ca8bb11dae76f3a61da99d9f5b9
- [`ui-refresh-overrides`](https://github.com/basil/jenkins/tree/ui-refresh-overrides): basil/jenkins@f7cf5dda62

I am requesting that we close this PR and file four _separate_ PRs based on each of the above branches to continue with integration. The reasoning for this is as follows:

- It is easier to do reviews when each logical unit of change is separate from the others. The reviewer only needs to keep in mind one logical change when reading the diff.
- It is easier to assess risk (including searching for usages in sources and binaries) and to do testing when each logical unit of change is separate from the others.
- In the event of a regression, it is easier to revert the invalid changes from one PR without touching the valid changes from the other PRs.

Of the four PRs in the split I have proposed, I am ready (or close to ready) to approve three of them. The `ui-refresh-overrides` change needs some more consideration. The nice thing about splitting things up is that we don't need to remain blocked on the first three changes while we consider the fourth. We can move faster.

Please note that I highly appreciate the cleanup effort you have done here. It is much needed, and it is not easy to do these cleanups. I should know, having done many similar backend cleanups myself in the Jenkins project and elsewhere. But unfortunately, with the benefits of these cleanups comes an associated risk of regression. My request to split up this PR is not intended to slow you down or to introduce additional bureaucracy. It is intended only to safeguard the stability of Jenkins core as we prepare for the forthcoming LTS release. On the contrary, I think we can move _faster_ with reviews and integrations if they are split up in this manner. I look forward to landing these cleanup improvements and many other improvements going forward.

**Comment by NotMyFault** [comment]
While I agree with your bullet points, I - and likely others too - tend to push several similar changes in a single PR to possibly not "spam" PRs. I can only speak for myself, but that somehow became a habit I'm used to from contributing to other projects 😅 

A note on the next LTS release; the baseline selection's due date and the RC creation of 2.332.3 are scheduled for next week Wednesday (March the 20th) and have both no mailing list post yet.
2.344, if selected, would be the last possible candidate these changes could go in, giving us a plenty of time to review them individually.

**Pull request closed without merging**
